src/olympia/addons/models.py

```diff
@@ -541,6 +546,32 @@ def from_upload(cls, upload, platforms, source=None, is_listed=True,
 
         return addon
 
+    @classmethod
+    def resolve_webext_translations(cls, data, upload):
+        """Resolve all possible translations from an add-on.
+
+        This returns a modified `data` dictionary accordingly with proper
+        translations filled in.
+        """
+        if not data.get('is_webextension') or not data.get('default_locale'):
+            # Don't change anything if we don't meet the requirements
+            return data
+
+        fields = ('name', 'homepage', 'summary')
+        messages = extract_translations(upload)
+
+        for field in fields:
+            data[field] = {
+                locale: resolve_i18n_message(
```

Comment:
You probably want to have `to_language(locale)` instead of `locale` here: we need to make sure we use the right format before saving in db. I think this is the best place to do it, if you modified `resolve_i18n_message` to do it it'd be messy cause you'd have to juggle between the locale format in the package and the one we want in the db.

Add a test, too...